conans/util/files.py

```diff
@@ -111,6 +113,27 @@ def save(path, content, append=False):
         handle.write(to_file_bytes(content))
 
 
+def mkdir_tmp():
```

Comment:
Naming is a bit weird, ``mkdir_tmp()`` vs ``tmp_file``?

Reply:
Also, some example use case of this utility for conanfiles?
I don't like very much the idea of conan recipes creating temporary files outside their package folder in the local cache.

Reply:
Ok, I'll remove them.

Reply:
I removed from tools. I'll rename our internal tools now :P

Reply:
`make_tmp_dir` and `new_tmp_file` is ok? or where was the weirdness? Notice that now are in our files not public module.